src/main/kotlin/com/beust/klaxon/Render.kt

```diff
@@ -10,7 +10,20 @@ private fun <A: Appendable> A.renderString(s: String): A {
             '\\' -> append(ch).append(ch)
             '\n' -> append("\\n")
             '\r' -> append("\\r")
-            else -> append(ch)
+            '\t' -> append("\\t")
+            '\b' -> append("\\b")
+            '\u000c' -> append("\\f")
+            else -> {
+                if(isPrintableUnicode(ch)){
+                    append("\\u")
+                    append(Integer.toHexString(ch.toInt().shr(12).and(0x000F)))
```

Comment:
Lot of repetition here, should create a function that does everything:

```
fun toUnicode(ch: Char, code: Int) = Integer.toHexString(ch.toInt().shr(code).and(0x000F))
```
and then:
```
append(toUnicode(ch, 12)).append(toUnicode(ch, 8)...
```